backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraHealthCheck.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra.utils;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.core.healthcheck.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.Session;
+
+/**
+ * Health check for the Cassandra backend.
+ *
+ */
+public class CassandraHealthCheck implements HealthCheck {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraHealthCheck.class);
+    private static final ComponentName COMPONENT_NAME = new ComponentName("Cassandra backend");
+    private static final String SAMPLE_QUERY = "SELECT NOW() FROM system.local";
+
+    private final Session session;
+
+    @Inject
+    public CassandraHealthCheck(Session session) {
+        this.session = session;
+    }
+
+    @Override
+    public ComponentName componentName() {
+        return COMPONENT_NAME;
+    }
+
+    @Override
+    public Result check() {
+        try {
+            // execute a simple query to check if cassandra is responding
+            // idea from: https://stackoverflow.com/questions/10246287
+            session.execute(SAMPLE_QUERY);
```

Comment:
Actually I was thinking of the query more like the `SELECT 1 FROM dual` like SQL. I just want to check if the connection is open and the Backend is responding not with an error. If you have a idea which data could be checked in the Cassandra-DB that is always available and makes sense, this can be changed.

Reply:
very sad that I don't know much about cassandra, if people agree with your solution, then it's fine to me :slightly_smiling_face:

Reply:
I do not come up with better ideas.

IMO we should avoid application data to keep the solution generic (could be reused on other projects) and avoid coupling.

Thus checking system tables seems good to me.